visualization/resources/web/vis/chartWizard/genericChartPanel.js

```diff
@@ -663,29 +664,38 @@ Ext4.define('LABKEY.ext4.GenericChartPanel', {
 
     renderDataGrid : function(renderTo)
     {
-        var urlParams = LABKEY.ActionURL.getParameters(this.baseUrl);
-        var filterUrl = urlParams['filterUrl'];
-        var userFilters, wpConfig, wp;
+        var filters = [];
+        var removableFilters = this.userFilters;
+        var dataRegion = LABKEY.DataRegions[this.dataRegionName];
 
-        if (this.userFilters)
-            userFilters = this.userFilters;
+        if (dataRegion !== undefined)
+        {
+            // If the region exists, then apply it's user filters as immutable filters to the QWP.
+            // They can be mutated on the data region directly.
+            filters = this.getDataRegionFilters();
+        }
         else
-            userFilters = LABKEY.Filter.getFiltersFromUrl(filterUrl, this.dataRegionName);
+        {
+            // If the region does not exist, then apply it's filters from the URL
+            // and allow them to be mutated on the QWP.
+            removableFilters = removableFilters.concat(this.getDataRegionFilters());
+        }
 
-        var userSort = LABKEY.Filter.getSortFromUrl(filterUrl, this.dataRegionName);
+        var userSort = LABKEY.Filter.getSortFromUrl(this.getFilterURL(), this.dataRegionName);
 
-        this.currentFilterStr = this.createFilterString(userFilters);
+        this.currentFilterStr = this.createFilterString(removableFilters);

```

Comment:
since `this.currentFilterStr` is used to determine if the chart needs to be re-rendered when the user switches between the "View Data" and "View Chart" button options, I think this needs to use the `filters` instead of `removableFilters`:
```suggestion
        this.currentFilterStr = this.createFilterString(filters);
```

Reply:
It ended up being even more subtle than this. I've updated this to include all filters since the user could modify any of these filters (e.g. either on the Data Region directly or on the QWP directly). See 1ba0ae4.